UofT-DSI | <SQL: Module 1-3> - Assignment 1

c5ho opened the pull request:
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
This is a submission of the SQL Modules - Assignment 1

## What did you learn from the changes you have made?
Basic SQL concepts such as database architecture, simple queries using SELECT, conditions, aggregate functions, joins, temp tables, datetime, etc.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
No other approaches were considered.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
I did run into some initial troubles using SQLite on WSL but that was resolved.

## How were these changes tested?
All queries were tested within DB Browser for SQLite, and a sample of the data returned was examined.

## A reference to a related issue in your repository (if applicable)

## Checklist
- [x] ] I can confirm that my changes are working as intended

c5ho committed:
Add Assignment 1, Section 1 Logical data model of Customer/Customer Purchases

c5ho committed:
Complete Assignment 1 Section 2

c5ho committed:
Complete Assignment 1 Section 3

l-mack reviewed (changes requested):
@c5ho Please review the feedback on your assignment below. Based on the comments provided, please revise and resubmit the indicated questions:

- Join Question 1: The question asks you to sort the result by vendor_name, then market_date, have you addressed this requirement? 
Aggregate Question 2: Look carefully at your GROUP BY clause. You're grouping by customer_last_name only, but selecting customer_first_name as well. Think about what happens when multiple customers share the same last name but have different first names.

c5ho committed:
Fixed JOIN 1., AGGREGATE 2., edited formatting and made more verbose where possible

c5ho commented:
@l-mack Thanks for the feedback.

Join Question 1: Added ORDER BY criteria
Aggregate Question 2: Added GROUP BY customer_id, customer_last_name, customer_first_name to ensure no records are missed if customers have the same first name and last name.

l-mack reviewed (approved):
Great job! 30/30
